cmd/task/list/cmd.go

```diff
@@ -12,15 +12,271 @@ import (
 	"fmt"
 	"os"
 	"strings"
-	"text/tabwriter"
 
+	"github.com/charmbracelet/lipgloss"
+	"github.com/charmbracelet/lipgloss/table"
 	"github.com/datarobot/cli/internal/task"
+	"github.com/datarobot/cli/tui"
 	"github.com/spf13/cobra"
 )
 
+// Category represents a human-readable task category
+type Category struct {
+	Name     string
+	Tasks    []task.Task
+	Priority int // Lower numbers appear first
+}
+
+// categoryStyles defines the styling for different category types
+var (
+	quickStartStyle = lipgloss.NewStyle().
+			Foreground(tui.DrGreen).
+			Bold(true)
+
+	buildingStyle = lipgloss.NewStyle().
+			Foreground(tui.DrPurple).
+			Bold(true)
+
+	testingStyle = lipgloss.NewStyle().
+			Foreground(tui.DrYellow).
+			Bold(true)
+
+	deploymentStyle = lipgloss.NewStyle().
+			Foreground(tui.DrIndigo).
+			Bold(true)
+
+	otherStyle = lipgloss.NewStyle().
+			Foreground(tui.DrPurpleLight).
+			Bold(true)
+)
+
+// getCategoryStyle returns the appropriate style for a category name
+func getCategoryStyle(categoryName string) lipgloss.Style {
+	switch {
+	case strings.Contains(categoryName, "Quick Start"):
+		return quickStartStyle
+	case strings.Contains(categoryName, "Building"):
+		return buildingStyle
+	case strings.Contains(categoryName, "Testing"):
+		return testingStyle
+	case strings.Contains(categoryName, "Deployment"):
+		return deploymentStyle
+	default:
+		return otherStyle
+	}
+}
+
+// getTaskCategory determines category based on task suffix
+func getTaskCategory(suffix string) *Category {
+	lowerSuffix := strings.ToLower(suffix)
+
+	if strings.Contains(lowerSuffix, "dev") || strings.Contains(lowerSuffix, "install") {
+		return &Category{Name: "🚀 Quick Start", Priority: 1}
+	}
+
+	if strings.Contains(lowerSuffix, "build") || strings.Contains(lowerSuffix, "docker") {
+		return &Category{Name: "🏗️  Building", Priority: 3}
+	}
+
+	if strings.Contains(lowerSuffix, "test") || strings.Contains(lowerSuffix, "lint") || strings.Contains(lowerSuffix, "check") {
+		return &Category{Name: "🧪 Testing & Quality", Priority: 4}
+	}
+
+	if strings.Contains(lowerSuffix, "deploy") || strings.Contains(lowerSuffix, "migrate") {
+		return &Category{Name: "� Deployment", Priority: 5}
+	}
+
+	return nil
+}
+
+// isCommonTask checks if a task is commonly used
+func isCommonTask(suffix string) bool {
+	lowerSuffix := strings.ToLower(suffix)
+	commonSuffixes := []string{"dev", "build", "test", "install", "deploy", "lint"}
+
+	for _, common := range commonSuffixes {
+		if strings.Contains(lowerSuffix, common) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// categorizeTask determines the appropriate category for a task
+func categorizeTask(t task.Task, showAll bool) *Category {
+	name := t.Name
+
+	// Root-level tasks
+	if !strings.Contains(name, ":") {
+		if cat := getTaskCategory(name); cat != nil {
+			return cat
+		}
+
+		return &Category{Name: "🚀 Quick Start", Priority: 1}
```

Comment:
_May_ be a little pre-optimization but should we put string `"🚀 Quick Start"` in a variable in this file so it's only needed to update in one place if/when someone wants to alter it?